driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -637,12 +672,14 @@ public SimpleEncodingFilter(final String fieldName, final TItem value) {
     private static <TItem> void encodeValue(final BsonDocumentWriter writer, final TItem value, final CodecRegistry codecRegistry) {
         if (value == null) {
             writer.writeNull();
+        } else if (value instanceof Bson) {
+            ((Encoder) codecRegistry.get(BsonDocument.class)).encode(writer,
+                    ((Bson) value).toBsonDocument(BsonDocument.class, codecRegistry), EncoderContext.builder().build());
         } else {
             ((Encoder) codecRegistry.get(value.getClass())).encode(writer, value, EncoderContext.builder().build());
         }
     }
 
-    // WIP, to be used later by not/nor methods
     private static class NotFilter implements Bson {
```

Comment:
This implementation is completely different from the .NET version. What is the reason for that?

It makes it hard to compare whether it's doing the same thing or not.